--- src/CMakeLists.txt	2023-12-07 03:53:02.257526792 +0000
+++ src/CMakeLists.txt	2023-12-07 03:52:50.833240906 +0000
@@ -5,7 +5,13 @@
 file(GLOB_RECURSE headers *.h)
 
 add_library(vaccel SHARED ${headers} ${sources})
-target_compile_options(vaccel PUBLIC -Wall -Wextra -Werror -pthread)
+
+if (ENABLE_COVERAGE)
+    target_compile_options(vaccel PUBLIC -Wall -Wextra -Werror -pthread --coverage)
+else()
+    target_compile_options(vaccel PUBLIC -Wall -Wextra -Werror -pthread)
+endif()
+
 target_include_directories(vaccel PRIVATE ${CMAKE_SOURCE_DIR}/src)
 target_include_directories(vaccel PRIVATE ${CMAKE_SOURCE_DIR}/src/include)
 target_include_directories(vaccel PRIVATE ${CMAKE_BINARY_DIR})
@@ -13,7 +19,13 @@
 target_include_directories(vaccel PRIVATE ${CMAKE_SOURCE_DIR}/src/include)
 set_property(TARGET vaccel PROPERTY LINK_FLAGS "-pthread")
 set_property(TARGET vaccel PROPERTY C_STANDARD 11)
-target_link_libraries(vaccel slog)
+
+if (ENABLE_COVERAGE)
+	target_link_libraries(vaccel slog --coverage)
+else()
+	target_link_libraries(vaccel slog)
+endif()
+
 
 find_package(Git)
